Core/UIViewControllerExtension.swift

```diff
@@ -132,6 +132,14 @@ extension Core.Link: UIActivityItemSource {
 
     public func activityViewController(_ activityViewController: UIActivityViewController,
                                        itemForActivityType activityType: UIActivity.ActivityType?) -> Any? {
+        
+        // We don't want to save localPath to favorites or bookmarks
+        if let localPath = localPath,
+           activityType != .saveBookmarkInDuckDuckGo,
+           activityType != .saveFavoriteInDuckDuckGo {
+        
+            return localPath
+        }
```

Comment:
This change impacts the list of activities shown. I think similar check _"if we have the local file downloaded"_ needs to be done also in the `func activityViewControllerPlaceholderItem`. Please verify.